src/HttpClientInterception/HttpClientInterceptorOptions.cs

```diff
@@ -387,10 +392,23 @@ public virtual HttpClient CreateHttpClient(HttpMessageHandler innerHandler = nul
         /// </summary>
         /// <param name="method">The HTTP method.</param>
         /// <param name="uri">The HTTP request URI.</param>
+        /// <param name="ignoreQueryString">If true create a key without any query string but with an extra string to disambiguate.</param>
         /// <returns>
         /// A <see cref="string"/> to use as the key for the interceptor registration.
         /// </returns>
-        private static string BuildKey(HttpMethod method, Uri uri) => $"{method.Method}:{uri}";
+        private static string BuildKey(HttpMethod method, Uri uri, bool ignoreQueryString = false)
+        {
+            if (ignoreQueryString)
+            {
+                var uriWithoutQueryString = uri == null ? null : new UriBuilder(uri) { Query = string.Empty }.Uri;
```

Comment:
Is it possible for `uri` to be `null`?

Reply:
One of the existing tests explicitly tests a "blank" WebRequest.  I can't see it happening in real life.

Reply:
That's fine. I just wasn't sure whether or not it was guarded against somewhere else so would have been an impossible condition to hit.